ocaml/xapi/xapi_sr.ml

```diff
@@ -124,6 +124,34 @@ let scanning_thread () = Debug.with_thread_named "scanning_thread" (fun () ->
 		done)
 	) ()
 
+let sr_cache = ref []
+
+let update_physical_utilisation ~__context =
+	let srs = Helpers.get_all_plugged_srs ~__context in
+	(* Remove SRs from sr_cache if they are no more available *)
+	let remove_srs = List.filter (fun sr -> (List.mem sr srs) = false ) !sr_cache in
+	sr_cache := List.filter (fun sr -> (List.mem sr remove_srs) = false ) !sr_cache;
+	(* Checks whether SR is sr_master and has SR_STATS capability *)
+	let is_stats_sr ~__context sr =
+		let sr_record = Db.SR.get_record_internal ~__context ~self:sr in
+		if (Smint.(has_capability Sr_stats (Xapi_sr_operations.features_of_sr ~__context sr_record)) &&
+			(Helpers.i_am_srmaster ~__context ~sr))
+			then true else false
+		in
```

Comment:
This `in` should either be on the end of the previous line or aligned with the `let`.